### PR TITLE
Update classic `distill` function

### LINKED ISSUE #39: [BUG] REDCapTidieR Duplicating Similarly Named Vars Across Tables
# Current Behavior
When testing the classic REDCap database, it was found that variables were being duplicated between  the `data_field_types` and `text_input_validation_types` instruments due to a `starts_with` selector in one of the cleaning sub functions. This caused the `text` variable in one instrument to pick up multiple `text_*` variables in another instrument.

The offending code lines can be found below:
https://github.com/CHOP-CGTDataOps/REDCapTidieR/blob/5d25913e3ad21733322eebef68eb8600d4bf488e/R/clean_redcap.R#L145-L147

Tests will also need to be updated.

## Proposed Solution:
In the longitudinal functions, this was handled by changing `starts_with()` to `all_of()`, but seems to not have carried over to the classic functions. 

## How to Reproduce the Bug:

```r
library(REDCapTidieR) 

out <- read_redcap_tidy(redcap_uri = Sys.getenv("REDCAP_URI"),
                                 token = Sys.getenv("REDCAPTIDIER_CLASSIC_API")) %>%
           extract_tables(c("data_field_types", "text_input_validation_types"))
```

### Checklist
Before submitting this issue, please check and verify below that the submission meets the below criteria:

- [x] The issue is atomic
- [x] The issue description is documented
- [x] The issue title describes the problem succinctly
- [x] Developers are assigned to the issue
- [x] Labels are assigned to the issue

### PR DESCRIPTION
**Pull request by rsh52**
# Description
This PR addresses a bug caused by use of `tidyselect::starts_with` in one of the lower-level `distill` functions for classic databases. I believe this must have been an oversight, since we implement `all_of` at the same location in longitudinal `distill` sub functions.

After implementation, I verified that the output of `read_redcap_tidy` for our classic test database behaves as expected and `test_*` variables are no longer crossing over between table extractions.

# Proposed Changes 
List changes below in bullet format:

- Change `starts_with` to `all_of` in `distill` classic sub function
- Update `create_test_data` internal script for classic database to reflect what `distill` sees at the point its called
   - Create new `classic` database output for tests

### Notes:
Looks like worldclockapi already got restored and the local and GHA checks passed without notes.

~Test suite and R CMD Check run fine, but there is a new note:~

```r
❯ checking for future file timestamps ... NOTE
  unable to verify current time

0 errors ✔ | 0 warnings ✔ | 1 note ✖
```

~It looks like according to [this SO post](https://stackoverflow.com/questions/63613301/r-cmd-check-note-unable-to-verify-current-time) that it's related to [worldclockapi.com](http://worldclockapi.com/) which is currently down, so hopefully this resolves itself soon.~

image

### Issue Addressed
**Closes #39**

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] New/revised functions have associated tests
   - No new tests have been added, I didn't want to rely on checking for specific var names since we may change the database in the future
- [x] New/revised functions use appropriate naming conventions
- [x] New/revised functions don't repeat code
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [x] I checked that new files follow naming conventions and are in the right place
- [x] I checked that documentation is complete, clear, and without typos
- [x] I added/edited comments to explain "why" not "how"
- [x] I checked that all new variable and function names follow naming conventions
- [x] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [x] I checked that new tests address important edge cases

### CODE REVIEW

**Comment by rsh52** [comment]
Ah, looks like the worldclockapi already got restored and resolved anyway!

**Review by ezraporter — COMMENTED** [review]
@rsh52 I'm wondering if there's a test we can add to enforce the expectation that `distill_nonrepeat_table(form_name = "some_form")` *doesn't* return a result with field names from elsewhere in the metadata. My quick attempt which could be inserted in the [`distill_nonrepeat_table tibble contains expected columns and data types for all REDCap field types`](https://github.com/CHOP-CGTDataOps/REDCapTidieR/blob/5d25913e3ad21733322eebef68eb8600d4bf488e/tests/testthat/test-clean_redcap.R#L35) test:

```r
out <- distill_nonrepeat_table(form_name = "data_field_types",
                                 db_data = db_data_classic,
                                 db_metadata = db_metadata_classic)

  field_names_from_other_forms <- db_metadata_classic %>%
    filter(form_name != "data_field_types", field_name != "record_id") %>%
    pull(field_name)

  expect_true(!any(field_names_from_other_forms %in% names(out)))
```

**Comment by rsh52** [comment]
> ```r
> field_names_from_other_forms <- db_metadata_classic %>%
>     filter(form_name != "data_field_types", field_name != "record_id") %>%
>     pull(field_name)
> 
>   expect_true(!any(field_names_from_other_forms %in% names(out)))
> ```

I like that this is a very succinct sanity check. May fail if we ever do any database changes, but it'll be a pretty loud fail and easy to determine.

**Comment by skadauke** [comment]
Not sure I follow the above comments - do we want to add this code to the test suite to update this PR?

**Comment by rsh52** [comment]
> Not sure I follow the above comments - do we want to add this code to the test suite to update this PR?

Yes I think so, I will add today I just didn't get around to it yesterday. Will also be adding a small fix we found where `check_*` functions are being exported by `roxygen` tags (just need to remove `@export` from their documentation)

**Review by ezraporter — APPROVED** [review]

File: tests/testthat/test-clean_redcap.R [hunk 1]
```diff
@@ -38,6 +38,14 @@ test_that("distill_nonrepeat_table tibble contains expected columns and data typ
                                  db_data = db_data_classic,
                                  db_metadata = db_metadata_classic)
 
+  # Check to ensure forms do not appear from other forms
```
Comment (skadauke) [inline tests/testthat/test-clean_redcap.R:41]: ... that **field names** do not appear from other forms?

**Review by skadauke — APPROVED** [review]
Take a look at my one comments. Looks good to merge to me.
Thx!
